docker/install-llvm-mlir.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+apt remove -y --purge --auto-remove cmake
+apt update && \
+   apt install -y software-properties-common lsb-release && \
+   apt clean all
+wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
+apt-add-repository "deb https://apt.kitware.com/ubuntu/ $(lsb_release -cs) main"
+apt update
+apt install -y kitware-archive-keyring
+rm /etc/apt/trusted.gpg.d/kitware.gpg
+apt update
```

Comment:
This is the best way to install cmake on a C++-only build system, but all of this can potentially be replaced by `pip install cmake` and pull the latest version to the active Python env and be on PATH.

What are your thoughts on this replacement?

Reply:
Ah, that sounds excellent. I will test it out.